packages/kbn-server-http-tools/src/get_server_options.ts

```diff
@@ -27,6 +27,10 @@ export function getServerOptions(config: IHttpConfig, { configureTLS = true } =
   const options: ServerOptions = {
     host: config.host,
     port: config.port,
+    // manually configuring the listener
+    listener: getServerListener(config, { configureTLS }),
+    // must set to true when manually passing a TLS listener, false otherwise
+    tls: configureTLS && config.ssl.enabled,
```

Comment:
This is the concrete change of configuration introduced by this PR

Reply:
Probably not in the scope of this PR, but I'm wondering why we have a `configureTLS` parameter. We don't seem to have any use case where we set a `false` value for it.

Perhaps I missed something, but I have a feeling that it should either be removed, or surfaced and put together with the rest of `IHttpConfig`.